### PR TITLE
Create first version of connectors helm chart

### PR DESCRIPTION
**Pull request by RafalKorepta**
Next step would be to create tests and Github action pipeline.

### CODE REVIEW

File: charts/connectors/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            matchLabels:  {{ include "connectors-pod-labels" $root | nindent 14 }}
+          maxSkew: {{ $spread.maxSkew }}
+          topologyKey: {{ $spread.topologyKey }}
+          whenUnsatisfiable: {{ $spread.whenUnsatisfiable }}
+    {{- end }}
+      volumes:
+      {{- if and .Values.auth.sasl.userName .Values.auth.sasl.mechanism .Values.auth.sasl.secretRef }}
+        - name: rc-credentials
+          secret:
+            defaultMode: 292
```
Comment (joejulian) [inline charts/connectors/templates/deployment.yaml:202]:
```suggestion
            defaultMode: 0o444
```

File: charts/connectors/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            secretName: {{ .Values.auth.sasl.secretRef }}
+      {{- end }}
+        - emptyDir:
+            medium: Memory
+            sizeLimit: {{ .Values.deployment.connectTmpSizeLimit }}
+          name: rp-connect-tmp
```
Comment (joejulian) [inline charts/connectors/templates/deployment.yaml:208]: Should this be templated? I can imagine users that don't want ram disks when they have nvme disks.

File: charts/connectors/values.yaml [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      status: _internal_connectors_status
+  groupID: connectors-cluster
+
+# -- Authentication settings.
+# For details,
+# see the [SASL documentation](https://docs.redpanda.com/docs/manage/kubernetes/security/sasl-kubernetes/).
+auth:
+  sasl:
+    # -- The authentication mechanism to use for the superuser. Options are `scram-sha-256` and `scram-sha-512`.
+    mechanism: scram-sha-256
```
Comment (joejulian) [inline charts/connectors/values.yaml:97]: we default to 512 in the redpanda chart, should we do the same here?

**Review by joejulian — APPROVED** [review]
Few nits, but otherwise LGTM.

File: charts/connectors/templates/deployment.yaml (hunk 2 above)
Comment (RafalKorepta) [reply charts/connectors/templates/deployment.yaml:208]: ram disks is quicker than nvme disks. I can templated it sure.

File: charts/connectors/values.yaml (hunk 3 above)
Comment (RafalKorepta) [reply charts/connectors/values.yaml:97]: sure I will switch to 512

File: charts/connectors/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                status.storage.redpanda.remote.write={{ .Values.connectors.storage.remote.write.status }}
+                offset.storage.replication.factor={{ .Values.connectors.storage.replicationFactor.offset }}
+                config.storage.replication.factor={{ .Values.connectors.storage.replicationFactor.config }}
+                status.storage.replication.factor={{ .Values.connectors.storage.replicationFactor.status }}
+                producer.linger.ms={{ .Values.connectors.producerLingerMS }}
+                producer.batch.size={{ .Values.connectors.producerBatchSize }}
+                config.providers=file,secretsManager
+                config.providers.file.class=org.apache.kafka.common.config.provider.FileConfigProvider
+                {{- if .Values.connectors.secretManager.enabled }}
+                config.providers.secretsManager.class=com.github.jcustenborder.kafka.config.aws.SecretsManagerConfigProvider
```
Comment (tomasz-sadura) [inline charts/connectors/templates/deployment.yaml:109]: this is for AWS only, for GCP config is different. Please also note the providers were forked and the classes changed:

- `config.providers.secretsManager.class=com.redpanda.connectors.config.provider.aws.SecretsManagerConfigProvider` for AWS

- `config.providers.secretsManager.class=com.redpanda.connectors.config.provider.gcp.SecretManagerConfigProvider` for GCP

**Pull request merged**